From 28c28eb11ac1696ce08c9c90e8a7a7759bc5c50b Mon Sep 17 00:00:00 2001
From: Christian Hergert <chergert@redhat.com>
Date: Tue, 28 Aug 2018 12:22:57 -0700
Subject: [PATCH] kernel-symbol: respect ENABLE_POLKIT setting

This fixes the compilation when we have sysprofd fully disabled.
That is not an ideal configuration, but keeping it working
ensures that some of the code is easy copy/paste'able in the
future for some situations.
---
 lib/symbols/sp-kernel-symbol.c | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/lib/symbols/sp-kernel-symbol.c b/lib/symbols/sp-kernel-symbol.c
index 8eb8087..14e026f 100644
--- a/lib/symbols/sp-kernel-symbol.c
+++ b/lib/symbols/sp-kernel-symbol.c
@@ -18,8 +18,13 @@
 
 #define G_LOG_DOMAIN "sp-kernel-symbol"
 
+#include "config.h"
+
 #include <gio/gio.h>
-#include <polkit/polkit.h>
+
+#ifdef ENABLE_POLKIT
+# include <polkit/polkit.h>
+#endif
 
 #include "sp-kallsyms.h"
 
@@ -87,6 +92,7 @@ type_is_ignored (guint8 type)
 static gboolean
 authorize_proxy (GDBusConnection *conn)
 {
+#ifdef ENABLE_POLKIT
   PolkitSubject *subject = NULL;
   GPermission *permission = NULL;
   const gchar *name;
@@ -113,6 +119,7 @@ authorize_proxy (GDBusConnection *conn)
 failure:
   g_clear_object (&subject);
   g_clear_object (&permission);
+#endif
 
   return FALSE;
 }
-- 
2.17.0

